File: projects/batfish/src/batfish/main/BatfishException.java
package batfish.main;

public class BatfishException extends RuntimeException {

   private static final long serialVersionUID = 1L;

   public BatfishException(String msg) {
      super(msg);
   }

   public BatfishException(String msg, Exception e) {
      super(msg, e);
   }

}

File: projects/batfish_scrubber/src/org/batfish/batfish_scrubber/batfish_exception.py
'''
Created on Jan 7, 2016

@author: arifogel
'''

class BatfishException(Exception):
    '''
    classdocs
    '''


    def __init__(self, message, cause = None):
        super(BatfishException, self).__init__(message + ",\n\tcaused by " + repr(cause).decode('string_escape'))
        self.cause = cause
        '''
        Constructor
        '''
        self.message = message
